workflow/executor.py

```diff
@@ -401,17 +402,22 @@ def run(self):
         rai_config = self.resource_manager.get_rai_config()
         steps_iter = peekable(self.steps)
         for step in steps_iter:
-            # `recover_step` option has higher priority than `recover` option
-            if self.config.recover_step and not recover_step_reached:
-                if step.name == self.config.recover_step:
-                    recover_step_reached = True
-                if not recover_step_reached:
-                    self.logger.info(
-                        f"Recovery... Skipping the step {step.name} (id='{step.idt}') till reach recovery step")
-                    continue
-            elif self.config.recover and step.state == WorkflowStepState.SUCCESS:
-                self.logger.info(f"Recovery... Skipping the successful step {step.name} (id='{step.idt}')")
+            if self.config.selected_steps and step.name not in self.config.selected_steps:
+                self.logger.info(f"Step {step.name} (id='{step.idt}') is not selected. Skipping..")
                 continue
+            else:
```

Comment:
do we want to make sure that if multiple steps passed they are continuous? i.e. if I pass ConfigureSources & Materialize only that would break the run on a new DB.

Reply:
There are cases when this is not true. 
i.e. 1+ materialization steps and you want to run 1st and export. In this case they are not continues.
I think it should be on user since workflow executor consider that all steps are independent.